### PR TITLE
[test visibility] Add option to automatically report logs within tests when using `winston`

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?

Add instrumentation to `winston` to automatically add a transport configuration that reports logs to datadog agentlessly, following the same instructions shown in the docs today for agentless logs: https://docs.datadoghq.com/logs/log_collection/nodejs/?tab=winston30#agentless-logging. This only happens if `LogSubmissionPlugin` is active, which only happens if `DD_AGENTLESS_LOG_SUBMISSION_ENABLED` is passed.

### Motivation
Same as we do for other languages like Java ([see docs](https://docs.datadoghq.com/tests/correlate_logs_and_tests?tab=cloudciprovideragentless#setup)), we want to be able to automatically report logs to Datadog's logs product if `DD_AGENTLESS_LOG_SUBMISSION_ENABLED` is passed. 

ℹ️ Only `winston` in `jest`, `mocha` and `cucumber` is supported at the moment.

```javascript
const { createLogger, format, transports } = require('winston')
const { expect } = require('chai')

const logger = createLogger({
  level: 'info',
  exitOnError: false,
  format: format.json(),
  transports: [
    new transports.Console()
  ]
})

describe('test', () => {
  it('should return true', () => {
    logger.log('info', 'Hello simple log!') // this log will be automatically submitted if `DD_AGENTLESS_LOG_SUBMISSION_ENABLED` is defined

    expect(true).to.be.true
  })
})
```

### Plugin Checklist

- [x] Unit tests.

### CODE REVIEW

File: packages/dd-trace/src/plugins/log_plugin.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ module.exports = class LogPlugin extends Plugin {
   configure (config) {
     return super.configure({
       ...config,
-      enabled: config.enabled && config.logInjection
+      enabled: config.enabled && (config.logInjection || config.ciVisAgentlessLogSubmissionEnabled)
```
Comment (juan-fernandez) [inline packages/dd-trace/src/plugins/log_plugin.js:57]: we also want to log injection if we're automatically submitting logs from test visibility

File: integration-tests/selenium/selenium.spec.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ const {
 } = require('../../packages/dd-trace/src/plugins/util/test')
 const { NODE_MAJOR } = require('../../version')
 
-const cucumberVersion = NODE_MAJOR <= 16 ? '9' : '10'
+const cucumberVersion = NODE_MAJOR <= 16 ? '9' : 'latest'
```
Comment (juan-fernandez) [inline integration-tests/selenium/selenium.spec.js:21]: unrelated fix

File: packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+const Plugin = require('../../plugins/plugin')
+const log = require('../../log')
+
+function getWinstonLogSubmissionParameters (config) {
+  const { site, service } = config
+
+  const defaultParameters = {
+    host: `http-intake.logs.${site}`,
+    path: `/api/v2/logs?dd-api-key=${process.env.DD_API_KEY}&ddsource=winston&service=${service}`,
```
Comment (nikita-tkachenko-datadog) [inline packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js:9]: I think it is safer to include the API key as a request header (`dd-api-key`), rather than as part of the query string.

Comment (nikita-tkachenko-datadog) [reply packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js:9]: Although taking a closer look at this, I am not sure if this is what `winston` allows.

File: packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    return defaultParameters
+  }
+
+  try {
+    const url = new URL(process.env.DD_AGENTLESS_LOG_SUBMISSION_URL)
+    return {
+      host: url.hostname,
+      port: url.port,
+      ssl: url.protocol === 'https:',
+      path: defaultParameters.path
```
Comment (nikita-tkachenko-datadog) [inline packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js:23]: Again, not sure if this is possible with `winston`, but gzipping the logs would make sense.

File: integration-tests/automatic-log-submission.spec.js
Comment (nikita-tkachenko-datadog) [inline integration-tests/automatic-log-submission.spec.js:1]: Does/should it test cases when the logs are written not by the test itself but by the logic that it covers?

Comment (nikita-tkachenko-datadog) [inline integration-tests/automatic-log-submission.spec.js:1]: Does/should it test cases when the logs are written not by the test itself but by the logic that it covers?

**Review by nikita-tkachenko-datadog — APPROVED** [review]

File: integration-tests/automatic-log-submission.spec.js
Comment (juan-fernandez) [reply integration-tests/automatic-log-submission.spec.js:1]: oh yeah, that's a good test. The log being in the test code or somewhere should be the same (what matters is the active span), but it's a good test, thanks 👍

File: packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js (hunk 3 above)
Comment (juan-fernandez) [reply packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js:9]: safer because there's less likelihood of the API being logged somewhere, right? 

It is possible to set headers, yeah. I was following the instructions shown by APM for the APM-logs integration, which are more tested: https://docs.datadoghq.com/logs/log_collection/nodejs/?tab=winston30#agentless-logging. In this case I think it's worth doing the change though

File: packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js (hunk 4 above)
Comment (juan-fernandez) [reply packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js:23]: this complicates the integration considerably and not even the APM-logs integration suggests it, so I'll keep it: https://docs.datadoghq.com/logs/log_collection/nodejs/?tab=winston30#agentless-logging

File: packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js (hunk 3 above)
Comment (nikita-tkachenko-datadog) [reply packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js:9]: > safer because there's less likelihood of the API key being logged somewhere
yep, this is what I had in mind

File: integration-tests/automatic-log-submission.spec.js
Comment (juan-fernandez) [reply integration-tests/automatic-log-submission.spec.js:1]: done in https://github.com/DataDog/dd-trace-js/pull/4762/commits/1857aedc7c0d446e4c408c91b997a9012d4f2874

File: packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js (hunk 3 above)
Comment (juan-fernandez) [reply packages/dd-trace/src/ci-visibility/log-submission/log-submission-plugin.js]: done in https://github.com/DataDog/dd-trace-js/pull/4762/commits/0d70afb92e060a5c11b08a5a1709d8f0e003e3d8

**Review by nikita-tkachenko-datadog — APPROVED** [review]
